crates/gen/src/tables/type_def.rs

```diff
@@ -181,6 +195,21 @@ impl TypeDef {
             .next()
     }
 
+    pub fn overridable_methods(&self) -> BTreeSet<&'static str> {
+        let mut methods = BTreeSet::new();
+
+        for interface in self
+            .interface_impls()
+            .filter(|interface| interface.is_overridable())
+        {
+            for method in interface.interface().resolve().methods() {
+                methods.insert(method.name());
+            }
+        }
+
+        methods
```

Comment:
If `fn methods()` doesn't return `+ '_` (it doesn't seem to need to because everything is moved?), this works as well:

```suggestion
        self.interface_impls()
            .filter(|interface| interface.is_overridable())
            .flat_map(|interface| {
                interface
                    .interface()
                    .resolve()
                    .methods()
                    .map(|method| method.name())
            })
            .collect()
```

Reply:
Yes, the `+ '_` is a holdover from when I was still returning rows with non-static references. There's probably a lot of opportunity to clean things up - a lot of places I'd like to return iterators but end up having to return a container.

Reply:
But I'll tackle this separately.

Reply:
Ah, good to know. Sounds like some nice cleanup :+1:

Reply:
Did you plan to pick this one up in (or now, after, since it's been merged) #796?

Reply:
Feel free to contribute. Otherwise, I'll get to it. I'm still running into a few issues with moving to iterators more broadly.

Reply:
#798. There doesn't seem to be much more opportunity to clean up local `let mut` + `for` + `push/insert` loops, it's pretty clean already :)